numpyro/infer/einstein/utils.py

```diff
@@ -1,37 +1,41 @@
-import jax
 import jax.numpy as jnp
 
 
 def posdef(m):
     """  Map a matrix to a positive definite matrix, where all eigenvalues are >= 1e-5. """
     mlambda, mvec = jnp.linalg.eigh(m)
-    if jnp.ndim(mlambda) >= 2:
-        mlambda = jax.vmap(lambda ml: jnp.diag(jnp.maximum(ml, 1e-5)), in_axes=tuple(range(jnp.ndim(mlambda) - 1)))(
-            mlambda)
-    else:
-        mlambda = jnp.diag(jnp.maximum(mlambda, 1e-5))
-    return mvec @ mlambda @ jnp.swapaxes(mvec, -2, -1)
+    mlambda = jnp.maximum(mlambda, 1e-5)
+    return (mvec * jnp.expand_dims(mlambda, -2)) @ jnp.swapaxes(mvec, -2, -1)
 
 
 def sqrth(m):
     """ Map a matrix to a positive definite matrix and square root it. """
     mlambda, mvec = jnp.linalg.eigh(m)
-    if jnp.ndim(mlambda) >= 2:
-        mlambdasqrt = jax.vmap(lambda ml: jnp.diag(jnp.maximum(ml, 1e-5) ** 0.5),
-                               in_axes=tuple(range(jnp.ndim(mlambda) - 1)))(mlambda)
-        msqrt = mvec @ mlambdasqrt @ jnp.swapaxes(mvec, -2, -1)
-    else:
-        mlambdasqrt = jnp.diag(jnp.maximum(mlambda, 1e-5) ** 0.5)
-        msqrt = mvec * mlambdasqrt * jnp.swapaxes(mvec, -2, -1)
```

Comment:
@OlaRonning `msqrt = mvec * mlambdasqrt * jnp.swapaxes(mvec, -2, -1)` should be `msqrt = (mvec * mlambdasqrt) @ jnp.swapaxes(mvec, -2, -1)`?

Reply:
Nice catch. Yes, that is correct!
``` python
>>> arr = np.array([[2,1],[1,5]])
>>> mlambda, mvec = np.linalg.eigh(arr)
>>> mvec * np.diag(mlambda) * mvec.T
array([[1.5547002 , 0.        ],
       [0.        , 4.85747583]])
>>> (mvec * mlambda) @ mvec.T
array([[2., 1.],
       [1., 5.]])
```
Won't update it as this PR fixes with cleaner code and [test](https://github.com/pyro-ppl/numpyro/pull/842/files/896cb36c16a127ff8c97cacebd0e00e43a92e3c7#diff-22c8c05fa7e59344b962bb707c5840651f9c21bd9404591e2e4c38545e58a9e1R97-R103).